File: submodules/engine/ios/ModuleCrypto/ModuleCrypto.m
#import <React/RCTBridgeModule.h>
#import <React/RCTBridge.h>


@interface RCT_EXTERN_MODULE(ModuleCrypto, NSObject)

RCT_EXTERN_METHOD(asymmetricEncrypt:(NSString *)value withKey:(BOOL *)isProduction
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)


RCT_EXTERN_METHOD(getRandomByteKey:(NSString *)value
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(symmetricEncrypt:(NSString *)value withKey:(NSString *)key
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(symmetricDecrypt:(NSString *)encrypted withKey:(NSString *)key
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: submodules/engine/ios/ModuleCrypto/ModuleCrypto.swift
import Foundation

@objc(ModuleCrypto)
class ModuleCrypto: NSObject {

    @objc(asymmetricEncrypt:withKey:withResolver:withRejecter:)
    func asymmetricEncrypt(value: String, isProduction: Bool, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
      
        let data = AsymmetricCryptography.encryptV2(value: value, isProduction: isProduction)
          resolve(data)
    }

  @objc(getRandomByteKey:withResolver:withRejecter:)
    func getRandomByteKey(value: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {

        let data = SymmetricCryptography.getRandomByteKey()
          resolve(data)
      }

    @objc(symmetricEncrypt:withKey:withResolver:withRejecter:)
    func symmetricEncrypt(value: String, key: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        
        let data = SymmetricCryptography.encryptV2(value: value, key: key)
          resolve(data)
        }

    @objc(symmetricDecrypt:withKey:withResolver:withRejecter:)
    func symmetricDecrypt(encrypted: String, key: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {

        let data = SymmetricCryptography.decryptV2(encrypted: encrypted, key: key)
          resolve(data)
        }
}
